File: android/app/src/main/java/com/tiagohs/cinema_history/helpers/utils/ColorUtils.kt
package com.tiagohs.cinema_history.helpers.utils

import android.graphics.drawable.ColorDrawable
import android.content.res.ColorStateList
import android.graphics.drawable.RippleDrawable
object ColorUtils {

    fun getPressedColorRippleDrawable(normalColor: Int, pressedColor: Int): RippleDrawable {
        return RippleDrawable(
            getPressedColorSelector(normalColor, pressedColor),
            getColorDrawableFromColor(normalColor),
            null
        )
    }

    fun getPressedColorSelector(normalColor: Int, pressedColor: Int): ColorStateList {
        return ColorStateList(
            arrayOf(
                intArrayOf(android.R.attr.state_pressed),
                intArrayOf(android.R.attr.state_focused),
                intArrayOf(android.R.attr.state_activated),
                intArrayOf()
            ),
            intArrayOf(pressedColor, pressedColor, pressedColor, normalColor)
        )
    }

    fun getColorDrawableFromColor(color: Int): ColorDrawable {
        return ColorDrawable(color)
    }

    fun getRandomColorAssets(): ColorAsset {
        val listOfColors = listOf(
            ColorAsset("red", "md_white_1000"),
            ColorAsset("pink", "md_white_1000"),
            ColorAsset("purple", "md_white_1000"),
            ColorAsset("deep_purple", "md_white_1000"),
            ColorAsset("blue", "md_white_1000"),
            ColorAsset("light_blue", "md_white_1000"),
            ColorAsset("cyan", "md_white_1000"),
            ColorAsset("teal", "md_white_1000"),
            ColorAsset("green", "md_white_1000"),
            ColorAsset("light_green", "md_black_1000"),
            ColorAsset("lime", "md_white_1000"),
            ColorAsset("yellow", "md_black_1000"),
            ColorAsset("amber", "md_black_1000"),
            ColorAsset("orange", "md_black_1000"),
            ColorAsset("deep_orange", "md_white_1000"),
            ColorAsset("brown", "md_white_1000"),
            ColorAsset("grey", "md_white_1000"),
            ColorAsset("blue_grey", "md_white_1000")
        )

        return listOfColors.random()
    }
}

data class ColorAsset(
    val colorName: String,
    val textColorName: String
)

File: ios/HistoryCinema/HistoryCinema/Infraestructure/Utils/ColorUtils.swift
//
//  ColorUtils.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 21/03/21.
//

import Foundation

class ColorUtils {
    
    static func getRandomColorAssets() -> ColorAsset {
        let listOfColors = [
            ColorAsset(colorName: "red", textColorName: "md_white_1000"),
            ColorAsset(colorName: "pink", textColorName: "md_white_1000"),
            ColorAsset(colorName: "purple", textColorName: "md_white_1000"),
            ColorAsset(colorName: "deep_purple", textColorName: "md_white_1000"),
            ColorAsset(colorName: "blue", textColorName: "md_white_1000"),
            ColorAsset(colorName: "light_blue", textColorName: "md_white_1000"),
            ColorAsset(colorName: "cyan", textColorName: "md_white_1000"),
            ColorAsset(colorName: "teal", textColorName: "md_white_1000"),
            ColorAsset(colorName: "green", textColorName: "md_white_1000"),
            ColorAsset(colorName: "light_green", textColorName: "md_black_1000"),
            ColorAsset(colorName: "lime", textColorName: "md_white_1000"),
            ColorAsset(colorName: "yellow", textColorName: "md_black_1000"),
            ColorAsset(colorName: "amber", textColorName: "md_black_1000"),
            ColorAsset(colorName: "orange", textColorName: "md_black_1000"),
            ColorAsset(colorName: "deep_orange", textColorName: "md_white_1000"),
            ColorAsset(colorName: "grey", textColorName: "md_white_1000"),
            ColorAsset(colorName: "blue_grey", textColorName: "md_white_1000")
        ]

        return listOfColors.randomElement() ?? ColorAsset(colorName: "red", textColorName: "md_white_1000")
    }
    
    static func getReferenceColorAsset(_ type: String) -> ColorAsset {
        switch type {
            case "Livro":
                return ColorAsset(colorName: "md_red_500", textColorName: "md_white_1000")
            case "Série":
                return ColorAsset(colorName: "md_deep_orange_500", textColorName: "md_white_1000")
            case "Filme":
                return ColorAsset(colorName: "md_purple_500", textColorName: "md_white_1000")
            case "Youtube":
                return ColorAsset(colorName: "md_green_500", textColorName: "md_white_1000")
            default:
                return ColorAsset(colorName: "md_blue_500", textColorName: "md_white_1000")
        }
    }
}
